roles/ssl_certificates/defaults/main.yml

```diff
@@ -0,0 +1,28 @@
+sites_using_ssl: "{{ wordpress_sites.values() | map(attribute='ssl') | selectattr('enabled', 'equalto', true) }}"
+sites_using_acme_ssl: "{{ sites_using_ssl | rejectattr('manual', 'defined') | map(attribute='acme') | map('default', ssl_defaults.acme )}}"
+sites_using_manual_ssl: "{{ sites_using_ssl | selectattr('manual', 'defined') }}"
```

Comment:
How should we write the new `wordpress_sites.yml`?

Most importantly, does `sites_using_acme_ssl`'s  `rejectattr('manual', 'defined')` correctly handle 3rd-party providers?

For example:

```
    ssl:
      enabled: true
      stapling_enabled: false
      provider: cloudflare-origin-ca
```

Reply:
>  Question 1: How should we write the new wordpress_sites.yml?

For the standard Let's Encrypt case, nothing changes if you want defaults. But here's what the implementation would expect:

```yaml
ssl:
  enabled: true
  acme:
    challenge:
      type: http-01
```

I structured it this way so that DNS challenges could be supported in the future:

```yaml
ssl:
  enabled: true
  acme:
    challenge:
      type: dns-01
      options:
        some_value: foo
```

Manual:

```yaml
ssl:
  enabled: true
  manual:
    certificate: path/to/cert
    key: path/to/key
```

> Question 2: Does sites_using_acme_ssl's rejectattr('manual', 'defined') correctly handle 3rd-party providers?

This PR (as is) wouldn't be compatible with any existing 3P roles that relied on `provider` existing. But the new structure just relies on a dict key being defined:


```yaml
ssl:
  enabled: true
  cloudflare-origin-ca:
    stapling_enabled: false
```

Because of how the tasks are split up and imported dynamically right now, none of the `manual` or `acme tasks would run for the above `cloudflare-origin-ca` example (which is good). That should make it fairly simple for a 3P role to conditionally run its own tasks based on the key existing?